forward_command_controller/include/forward_command_controller/forward_command_controller.h

```diff
@@ -85,14 +86,24 @@ class ForwardCommandController: public controller_interface::Controller<T>
   }
 
   void starting(const ros::Time& time);
-  void update(const ros::Time& time, const ros::Duration& period) {joint_.setCommand(command_);}
+  void update(const ros::Time& time, const ros::Duration& period)
+  {
+    tmp_command_ = *(command_.readFromRT());
+    joint_.setCommand(tmp_command_);
+    
+  }
 
   hardware_interface::JointHandle joint_;
-  double command_;
+  realtime_tools::RealtimeBuffer<double> command_;
+  double tmp_command_; // pre-allocated memory that is re-used to set the realtime buffer
```

Comment:
`tmp_command_` does not have to be a class member, as it will be allocated on the stack as an automatic variable. You could just use it in the `update(...)` method as:
```c++
const double cmd = *(command_.readFromRT());
joint_.setCommand(cmd);
```

if the command was a more complex structure, involving variable-sized containers for instance, then preallocation would definitely make sense.